src/components/app/app.tsx

```diff
@@ -22,27 +25,47 @@ function App() {
         if (!json.success) {
           throw new Error('ошибка');
         }
-        setData(json.data);
+        setIngredientData(json.data);
       })
       .catch((error) => {
         console.error('Что то пошло не так... :(', error);
       })
-  }, [])
+  }, []);
+
+  const initialСartState: cartData = {
+    items: ingredientData,
+    totalPrice: 0,
+  }
+  function cartReducer(state: cartData, action: Ingredient[]): cartData {
+    let totalPrice = 0;
+    action.forEach((item: any) => {
+      totalPrice += item.price;
+    });
```

Comment:
Тут также для подсчета суммы можно использовать метод `reduce`
```suggestion
    const totlePrice = action.reduce((item: Ingredient) => {
      ...
    }, 0);
```